sec_certs_page/user/views.py

```diff
@@ -0,0 +1,464 @@
+    if form.validate_on_submit():
+        user_obj = User.get_by_email(form.email.data)
+        if user_obj:
+            token = User.generate_password_reset_token(user_obj.username)
+            reset_url = url_for('user.reset_password', token=token, _external=True)
+            
+            msg = Message(
+                subject="Password Reset - sec-certs",
+                recipients=[user_obj.email],
+                html=render_template("user/emails/reset_password.html.jinja2", 
+                                   user=user_obj, reset_url=reset_url),
+                sender=current_app.config.get('MAIL_DEFAULT_SENDER')
+            )
+            
+            try:
+                mail.send(msg)
+            except Exception as e:
+                pass  # Don't reveal if email exists
+        
+        flash("If the email exists in our system, you will receive password reset instructions.", "info")
+        return redirect(url_for("user.login"))
+    
+    return render_template("user/forgot_password.html.jinja2", form=form)
+
+
+@user.route("/reset-password/<token>", methods=["GET", "POST"])
+@register_breadcrumb(user, ".reset_password", "Reset Password")
+def reset_password(token):
+    if current_user.is_authenticated:
+        return redirect(url_for("index"))
+    
+    user_id = User.verify_token(token, 'password_reset')
+    if not user_id:
+        flash("Invalid or expired reset link.", "error")
+        return redirect(url_for("user.forgot_password"))
+    
+    user_obj = User.get(user_id)
+    if not user_obj:
+        flash("Invalid reset link.", "error")
+        return redirect(url_for("user.forgot_password"))
+    
+    form = PasswordResetForm()
+    if form.validate_on_submit():
+        user_obj.set_password(form.password.data)
+        flash("Password reset successfully! You can now log in.", "success")
+        return redirect(url_for("user.login"))
+    
+    return render_template("user/reset_password.html.jinja2", form=form, token=token)
+
+
+@user.route("/magic-link", methods=["GET", "POST"])
+@register_breadcrumb(user, ".magic_link", "Email Login")
+def magic_link():
+    if current_user.is_authenticated:
+        return redirect(url_for("index"))
+    
+    form = MagicLinkForm()
+    if form.validate_on_submit():
+        user_obj = User.get_by_email(form.email.data)
+        if user_obj and user_obj.email_confirmed:
+            token = User.generate_magic_link_token(user_obj.username)
+            login_url = url_for('user.magic_login', token=token, _external=True)
+            
+            msg = Message(
+                subject="Login Link - sec-certs",
+                recipients=[user_obj.email],
+                html=render_template("user/emails/magic_link.html.jinja2", 
+                                   user=user_obj, login_url=login_url),
+                sender=current_app.config.get('MAIL_DEFAULT_SENDER')
+            )
+            
+            try:
+                mail.send(msg)
```

Comment:
This and all of the other email sends should be handled asynchronously via dramatiq tasks. Just like how the notification emails are handled in the notifications blueprint.